### ISSUE TITLE
Blueprint routes fail on sails global config off

### ISSUE THREAD

**Issue by badunk** [balderdashy/sails#2730]
My config/globals.js:
```
module.exports.globals = false;
```
My UserController:
```
module.exports = {};
```
This is the error I get when I do `GET /user`:
```
Error:

 ReferenceError: sails is not defined
    at Object.module.exports.parseLimit (../node_modules/sails/lib/hooks/blueprints/actionUtil.js:284:25)
    at findRecords (../node_modules/sails/lib/hooks/blueprints/actions/find.js:42:22)
    at routeTargetFnWrapper (../node_modules/sails/lib/router/bind.js:179:5)
    at callbacks (../node_modules/sails/node_modules/express/lib/router/index.js:164:37)
    at param (../node_modules/sails/node_modules/express/lib/router/index.js:138:11)
    at pass (../node_modules/sails/node_modules/express/lib/router/index.js:145:5)
    at nextRoute (../node_modules/sails/node_modules/express/lib/router/index.js:100:7)
    at callbacks (../node_modules/sails/node_modules/express/lib/router/index.js:167:11)
```

That file seems to assume sails is going to be available globally.

**Comment by tjwebb** [comment]
What version of sails are you using?

**Comment by badunk** [comment]
I'm using 0.11.0

**Comment by ghaiklor** [comment]
@badunk it's because in global config you can set that you don't want expose sails to global. When you set `module.exports.globals = false` sails is not accessible via global context. Here the full config file:
```javascript
module.exports.globals = {
    /**
     * Expose the lodash installed in Sails core as a global variable
     */
    _: true,

    /**
     * Expose the async installed in Sails core as a global variable
     */
    async: true,

    /**
     * Expose the sails instance representing your app
     */
    sails: true,

    /**
     * Expose each of your app's services as global variables
     */
    services: true,

    /**
     * Expose each of your app's models as global variables
     */
    models: true
};
```

You can disable all but make sure that `sails: true` :smiley:

**Comment by badunk** [comment]
@ghaiklor oh, I know its because of that.  However, I would have thought a core module wouldn't have depended on the global sails if the config was documented as an option.

**Comment by ghaiklor** [comment]
@badunk yeah, seems like hardcoded pointer to sails - https://github.com/balderdashy/sails/blob/master/lib/hooks/blueprints/actionUtil.js#L284 . They are expecting that sails is accessible via global. If take attention that it's `actionUtil.js`, not a part of some hook, etc... so there is no any callback with sails argument. But there is should be some workaround to fix that.

**Comment by CWyrtzen** [comment]
Has anyone found a workaround? Is this a bug?

**Comment by tarlepp** [comment]
Yep the workaround is to fix ``´sails``` references to ```req._sails``` Seems like there is three "wrong" usages on that actionUtil.js file.

There is/was similar problems with ```_``` require.

**Comment by sailsbot** [comment]
Thanks for posting, @badunk.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by tarlepp** [comment]
@sailsbot closed this one too, but there isn't any information about actual fix of this issue?
